src/Plivo/Resources/Number/NumberInterface.php

```diff
@@ -59,15 +59,26 @@ public function getList($optionalArgs = [])
 
         $numbers = [];
 
-        foreach ($response->getContent()['objects'] as $number) {
-            $newNumber = new Number(
-                $this->client, $number, $this->pathParams['authId']);
+        $responseContents = $response->getContent();
+        if(!array_key_exists("error",$responseContents)){
+            foreach ($response->getContent()['objects'] as $number) {
+                $newNumber = new Number(
+                    $this->client, $number, $this->pathParams['authId']);
+
+                array_push($numbers, $newNumber);
+            }
+            return new ResourceList(
+                $this->client, $response->getContent()['meta'], $numbers);
+        } else {
```

Comment:
Lets first check the error and `return` the exception if the error exists

Lets only return `ResourceList` for non error